### ISSUE TITLE
Style/StructInheritance cop bugs on inline class definition

### ISSUE THREAD

**Issue by edouard-per-angusta** [rubocop/rubocop#9698]
The Style/StructInheritance cop bugs on inline class definition.

--------

## Expected behavior

If there a struct defined in the following format:

```ruby
class Foo < Struct.new(:bar, :baz); end
```

it should render the same error it renders when `end` is on a new line:

```
example.rb:3:13: C: [Correctable] Style/StructInheritance: Don't extend an instance initialized by Struct.new. Use a block to customize the struct.
class Foo < Struct.new(:bar, :baz)
            ^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## Actual behavior

Parser bugs:

```
1 error occurred:
An error occurred while Style/StructInheritance cop was inspecting /.../example.rb:3:0.
```

## Steps to reproduce the problem

Create a file with the code below and run rubocop on it:

```ruby
class Foo < Struct.new(:bar, :baz); end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.12.1 (using Parser 3.0.1.0, rubocop-ast 1.4.1, running on ruby 2.6.7 x86_64-darwin20)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.2.0
```
